### PR TITLE
Pass Pylint up to 2.4

### LINKED ISSUE #3087: Pylint fails with development branch and 2.16 branch
### Description
- Type: Bug
- Priority: Minor

---------------------------------------------------------------
## Bug

When running pylint version 1.8.3 (which is quite old), I get the following errors for the `development` branch:
```
******************************************************************
* check_python_files: Lint: Python scripts
* Thu Mar  5 14:21:14 UTC 2020
******************************************************************
Using config file /home/simbut01/work/mbedtls/mbedtls-development/.pylintrc
************* Module config
I: 41, 0: Locally disabling too-few-public-methods (R0903) (locally-disabled)
I:462, 0: Locally disabling wrong-import-position (C0413) (locally-disabled)
R:103, 8: Unnecessary "else" after "return" (no-else-return)
R:282,16: Redefining argument with the local name 'filename' (redefined-argument-from-local)
R:312, 8: Unnecessary "else" after "return" (no-else-return)
R:373, 4: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
I:462, 0: Useless suppression of 'wrong-import-position' (useless-suppression)
I: 30, 0: Suppressed 'too-few-public-methods' (from line 41) (suppressed-message)
************* Module check-files
I: 34, 0: Locally disabling no-member (E1101) (locally-disabled)
C: 53, 4: Missing method docstring (missing-docstring)
I: 55, 0: Suppressed 'no-member' (from line 34) (suppressed-message)
************* Module abi_check
I:427, 0: Locally disabling broad-except (W0703) (locally-disabled)
W: 85,16: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
W: 98,27: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
W:114,23: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
R:172, 8: Unnecessary "else" after "return" (no-else-return)
W:274,20: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
I:427, 0: Suppressed 'broad-except' (from line 427) (suppressed-message)
************* Module check-test-cases
C: 79, 0: Missing function docstring (missing-docstring)
C: 96, 0: Missing function docstring (missing-docstring)
************* Module generate_test_code
I:241, 0: Locally disabling no-member (E1101) (locally-disabled)
I:241, 0: Suppressed 'no-member' (from line 241) (suppressed-message)
************* Module test_config_script
R: 91, 4: Unnecessary "else" after "return" (no-else-return)
************* Module mbedtls_test
I: 41, 0: Locally disabling import-error (E0401) (locally-disabled)
I: 41, 0: Useless suppression of 'import-error' (useless-suppression)
************* Module test_generate_test_code
I: 25, 0: Locally disabling wrong-import-order (C0411) (locally-disabled)
I: 39, 0: Locally enabling wrong-import-order (C0411) (locally-enabled)
I:340, 0: Locally disabling unused-argument (W0613) (locally-disabled)
I: 25, 0: Useless suppression of 'wrong-import-order' (useless-suppression)
I:333, 0: Suppressed 'unused-argument' (from line 340) (suppressed-message)

-----------------------------------
Your code has been rated at 9.93/10
```

And the following errors for the `mbedtls-2.16` branch:
```
******************************************************************
* check_python_files: Lint: Python scripts
* Thu Mar  5 11:16:34 UTC 2020
******************************************************************
Using config file /home/simbut01/work/mbedtls/mbedtls-2.16/.pylintrc
************* Module check-files
I: 34, 0: Locally disabling no-member (E1101) (locally-disabled)
C: 53, 4: Missing method docstring (missing-docstring)
I: 55, 0: Suppressed 'no-member' (from line 34) (suppressed-message)
************* Module generate_test_code
I:241, 0: Locally disabling no-member (E1101) (locally-disabled)
I:241, 0: Suppressed 'no-member' (from line 241) (suppressed-message)
************* Module abi_check
I:427, 0: Locally disabling broad-except (W0703) (locally-disabled)
W: 85,16: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
W: 98,27: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
W:114,23: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
R:172, 8: Unnecessary "else" after "return" (no-else-return)
W:274,20: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
I:427, 0: Suppressed 'broad-except' (from line 427) (suppressed-message)
************* Module mbedtls_test
I: 41, 0: Locally disabling import-error (E0401) (locally-disabled)
I: 41, 0: Suppressed 'import-error' (from line 41) (suppressed-message)
************* Module test_generate_test_code
I: 25, 0: Locally disabling wrong-import-order (C0411) (locally-disabled)
I: 39, 0: Locally enabling wrong-import-order (C0411) (locally-enabled)
I:340, 0: Locally disabling unused-argument (W0613) (locally-disabled)
I: 25, 0: Useless suppression of 'wrong-import-order' (useless-suppression)
I:333, 0: Suppressed 'unused-argument' (from line 340) (suppressed-message)

------------------------------------------------------------------
Your code has been rated at 9.96/10 (previous run: 9.96/10, +0.00)
```

### LINKED ISSUE #3111: Pylint may invoke Python2 pylint
### Description
- Type: Bug
- Priority: Major

---------------------------------------------------------------
## Bug

Since https://github.com/ARMmbed/mbedtls/pull/3100, `all.sh check_python_files` fails on systems that have both Python 2 including `pylint` and Python 3 including `pylint3`. Python 2 is no longer upported, but it's too early to assume that everybody has uninstalled it.

For at least a few months, we should prefer `pylint3` to `pylint`.

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Until now we've only sanitized our Python scripts and set up our `.pylintrc` for the version of Pylint in Ubuntu 16.04 (Pylint 1.5.2). This pull request sanitizes our Python scripts for the latest version of Pylint. Fix #3087

I've tested locally that `check-python-files.sh` passes with Pylint 1.5.2, 1.8.3 and 2.4.4.

This pull request makes two other changes:

* In `all.sh`, run `check_python_files` systematically. The rationale for making it optional was that pylint wasn't available on our CI, but these days, it is.
* In `check-python-files.sh`, don't run `pylint` without checking that it isn't for Python 2. Fix #3111

Backports: 2.16 #3184 - 2.7 (partial) #3185

### CODE REVIEW

File: tests/scripts/all.sh [hunk 1]
```diff
@@ -1750,15 +1750,6 @@ component_test_zeroize () {
     unset gdb_disable_aslr
 }
 
-support_check_python_files () {
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply tests/scripts/all.sh:1753]: `support_XXX` is called if it exists, and if it doesn't exist that's equivalent to a function that always returns true. See the [documentation](https://github.com/ARMmbed/mbedtls/blob/13c95c4d74436937e653bd4b214fcc333ed6cc70/tests/scripts/all.sh#L66). So removing the function is what calls it to be unconditional. The previous commit only affects what version of python is called, not whether pylint is called.

**Comment by gilles-peskine-arm** [comment]
@ronald-cron-arm  Diff of the pylint 2.4.4 output before/after "Pylint: allow if-return-else-return":
```
< scripts/abi_check.py:172:8: R1705: Unnecessary "else" after "return" (no-else-return)
8,13d6
< ************* Module config
< scripts/config.py:103:8: R1705: Unnecessary "else" after "return" (no-else-return)
< scripts/config.py:320:8: R1705: Unnecessary "elif" after "return" (no-else-return)
< scripts/config.py:447:8: R1705: Unnecessary "elif" after "return" (no-else-return)
< ************* Module generate_psa_constants
< scripts/generate_psa_constants.py:232:8: R1705: Unnecessary "elif" after "return" (no-else-return)
30d22
< tests/scripts/test_config_script.py:91:4: R1705: Unnecessary "else" after "return" (no-else-return)
```

So I don't understand what you mean about "locally disabled" messages.

**Comment by gilles-peskine-arm** [comment]
I amended a commit to fix a [typo](https://github.com/ARMmbed/mbedtls/pull/3118#discussion_r405351225). No other changes.

File: tests/scripts/all.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronald-cron-arm) [reply tests/scripts/all.sh:1753]: Thanks for the pointer to the documentation. I understand how it works now.

**Pull request merged**